regression/esbmc-cpp/bug_fixes/1176_issue/main.cpp

```diff
@@ -0,0 +1,21 @@
+#include <cassert>
+
+struct dim3
+{
+    unsigned int x, y, z;
+    dim3(unsigned int vx = 1, unsigned int vy = 1, unsigned int vz = 1)
+    {
+      x = vx;
+      y = vy;
+      z = vz;
+    }
+};
+
+int main() 
+{
+  assert(dim3(3).x == 3);
```

Comment:
Is `assert(dim3(3).z == 1);` mimicking any syntax or semantics in CUDA OM that's blocking us? 

If not, I'd suggest close this PR, and probably better to add a representative test case in 1189 and just focus on that PR. 

It's probably not the best thing to do to overcomplicates the code based on some edge cases. This is going to affect our debugging, maintanence and future extensions.